src/components/DiseaseContainer.tsx

```diff
@@ -1,22 +1,20 @@
 import React, { Fragment, FunctionComponent } from "react";
-import useApi from "./UseApi";
-import DiseaseCard, { DiseaseData } from "./cards/DiseaseCard";
+import { RouteComponentProps, withRouter, useParams } from "react-router";
+import useApi from "./hooks/UseApi";
+import { DiseaseData } from "./cards/DiseaseCard";
 import { baseUrl } from "../config";
+import DiseaseCardCompact from "./cards/DiseaseCardCompact";
 
-const DiseaseContainer: FunctionComponent<{ id: string }> = ({ id }) => {
+const DiseaseContainer: FunctionComponent<
+  { id: string } & RouteComponentProps
+> = () => {
+  const { id } = useParams();
   const { data } = useApi<{ result: DiseaseData }>(`${baseUrl}/diseases/${id}`);
   if (!data) {
     return null;
   }
   const { result } = data;
-  return (
-    <Fragment>
-      <div className="page-header">
-        <h2>Disease</h2>
-      </div>
-      {result && <DiseaseCard data={result} />}
-    </Fragment>
-  );
+  return <Fragment>{result && <DiseaseCardCompact data={result} />}</Fragment>;
 };
 
-export default DiseaseContainer;
+export default withRouter(DiseaseContainer);
```

Comment:
Not sure you need `withRouter` here

Reply:
Also, in other places. Because I see you're using `react-router`'s hooks already. I'd say in files that were already using `withRouter` it's fine to keep it, but in updated/new files go for the hooks